Fix issue 427

nmummau opened the pull request:
This fixes issue #427 
Also included are is a net9 framework target, and some docker compose cleanup

alexeyzimarev commented:
I'd prefer to split those, especially when it comes to docker compose updates. No need to do it with this one, it's ok.

The reason is that I'd definitely accept the fix, but docker compose change seems discussable. Using `latest` could suddenly break because it will be non-deterministic.

Also, samples don't need two framework targets, one is enough. Using just .NET 9 is fine.

alexeyzimarev commented:
Also, where the actual fix? I only see the props file changed to add .NET 9 and the rest is changes in docker compose.

nmummau committed:
chore: bump samples to use net9.0

nmummau committed:
fix(mssql): change from_position to int

This change resolves a code smell comparing a BIGINT to an INT.
This is safe and the correct thing to do because SqlServerStreamSubscription PrepareCommand() already is handling parsing this to an int:

protected override SqlCommand PrepareCommand(SqlConnection connection, long start)
        => connection.GetStoredProcCommand(Schema.ReadStreamSub)
            .Add("@stream_id", SqlDbType.Int, _streamId)
            .Add("@stream_name", SqlDbType.NVarChar, _streamName)
            .Add("@from_position", SqlDbType.Int, (int)start + 1)
            .Add("@count", SqlDbType.Int, Options.MaxPageSize);

nmummau commented:
@alexeyzimarev I rebased and cleaned this up. It's now an isolated fix to fix issue #427, along with an update to the TargetFramework for the samples.
